Push to stable

vanhauser-thc opened the pull request:
There is a bug in afl-clang-fast pcguard (default for 10.0.1 and newer):

```
$ cat main.c
int main(void) { return 0; }
```
that would fail with `afl-clang-fast -pie main.c`

so we should push to stable.

vanhauser-thc committed:
sync (#886)

* Create FUNDING.yml

* Update FUNDING.yml

domenukk committed:
moved custom_mutator examples

domenukk committed:
unicorn speedtest makefile cleanup

domenukk committed:
fixed example location

vanhauser-thc committed:
fix qdbi

vanhauser-thc committed:
update util readme

WorksButNotTested committed:
Frida persistent (#880)

* Added x64 support for persistent mode (function call only), in-memory teest cases and complog

* Review changes, fix NeverZero and code to parse the .text section of the main executable. Excluded ranges TBC

* Various minor fixes and finished support for AFL_INST_LIBS

* Review changes

Co-authored-by: Your Name <you@example.com>

vanhauser-thc committed:
nits

vanhauser-thc committed:
fix frida mode

Zherya committed:
Integer overflow/underflow fixes in libdislocator (#889)

* libdislocator: fixing integer overflow in 'max_mem' variable and setting 'max_mem' type to 'size_t'

* libdislocator: fixing potential integer underflow in 'total_mem' variable due to its different values in different threads

WorksButNotTested committed:
Bumped warnings up to the max and fixed remaining issues (#890)

Co-authored-by: Your Name <you@example.com>

vanhauser-thc committed:
nits

vanhauser-thc committed:
frida mode - support non-pie

vanhauser-thc committed:
nits

vanhauser-thc committed:
nit

vanhauser-thc committed:
update grammar mutator

WorksButNotTested committed:
Fixes for aarch64, OSX and other minor issues (#891)

Co-authored-by: Your Name <you@example.com>

vanhauser-thc committed:
nits

vanhauser-thc committed:
nits

vanhauser-thc committed:
fix PCGUARD, build aflpp_driver with fPIC

WorksButNotTested committed:
Added representative fuzzbench test and test for libxml (#893)

* Added representative fuzzbench test and test for libxml

* Added support for building FRIDA from source with FRIDA_SOURCE=1

Co-authored-by: Your Name <you@example.com>

vanhauser-thc committed:
nits

domenukk commented on .gitmodules:
```diff
@@ -7,3 +7,7 @@
 	url = https://github.com/AFLplusplus/qemuafl
+[submodule "frida_mode/frida"]
+	path = frida_mode/frida
+	url = https://github.com/WorksButNotTested/frida.git
```
Shouldn't we depend on upstream frida istead?

vanhauser-thc replied:
current upstream frida has a breaking bug. this is not for release but so we can push to stable for the moment.

domenukk commented on docs/Changelog.md:
```diff
@@ -10,6 +10,7 @@ sending a mail to <afl-users+subscribe@googlegroups.com>.
 
 ### Version ++3.13a (development)
```
Add changelog for grammar mutatior?

vanhauser-thc replied:
ok

domenukk commented on utils/libdislocator/libdislocator.so.c:
```diff
@@ -500,19 +500,20 @@ size_t malloc_usable_size(const void *ptr) {
+    char *             tok;
+    unsigned long long mmem = strtoull(tmp, &tok, 10);
+    if (*tok != '\0' || errno == ERANGE || mmem > SIZE_MAX / 1024 / 1024)
+      FATAL("Bad value for AFL_LD_LIMIT_MB");
```
I've seen a commit in the history recently called "Dominik loves brackets" :P

vanhauser-thc replied:
but he doesn't love them so much he does the work ;) (this is from an external PR)

vanhauser-thc committed:
update changelog

hexcoder- committed:
typos

hexcoder- commented on frida_mode/include/complog.h:
```diff
@@ -0,0 +1,14 @@
+#ifndef _COMPLOG_H
```
In Frida all occurences of "CompLog" should be renamed to our previously used "CmpLog" to avoid confusion, please!
That is: in files, functions and documentation

vanhauser-thc replied:
good catch, removed files and renamed files etc.

hexcoder- reviewed (approved):
Please remove binary files (which could be rebuild) from the repository:
frida_mode/src/lib/lib
frida_mode/test/fuzzbench/fuzzer
frida_mode/test/libxml/xml

domenukk committed:
fixed potential double free in custom trim (#881)

domenukk committed:
error handling, freeing mem

vanhauser-thc committed:
frida: complog -> cmplog

vanhauser-thc committed:
fix statsd writing

vanhauser-thc committed:
let aflpp_qemu_driver_hook.so build fail gracefully

vanhauser-thc committed:
fix stdin trimming

WorksButNotTested committed:
Support for AFL_ENTRYPOINT (#898)

Co-authored-by: Your Name <you@example.com>

vanhauser-thc commented:
please have a close look at 86452cc959bd4b0d5fe6e60d0eefbc7848fe38e2 and make a few tests.
a user stumbled upon a bug which seems that resulted in no trimming being performed against targets that read from stdin.

hexcoder- committed:
remove the input file .cur_input at the end of the fuzzing, if AFL_TMPDIR is used

vanhauser-thc committed:
reverse push (#901)

* Create FUNDING.yml

* Update FUNDING.yml

* disable QEMU static pie

Co-authored-by: Andrea Fioraldi <andreafioraldi@gmail.com>

hexcoder- committed:
clarify that no modifications are required.

hexcoder- committed:
add new test for frida_mode (please review)

hexcoder- committed:
typos

hexcoder- committed:
fix persistent mode (64-bit)

hexcoder- committed:
set ARCH for linux intel 32-bit for frida-gum-devkit

hexcoder- committed:
prepare for 32-bit support (later)

domenukk committed:
not on qemu 3 anymore

domenukk committed:
unicorn mips fixes

devnexen committed:
instrumentation further move to C++11 (#900)

domenukk committed:
unicorn fixes

domenukk committed:
more unicorn fixes

realmadsci committed:
Fix memory errors when trim causes testcase growth (#881) (#903)

* Revert "fixed potential double free in custom trim (#881)"

This reverts commit e9d2f72382cab75832721d859c3e731da071435d.

* Revert "fix custom trim for increasing data"

This reverts commit 86a8ef168dda766d2f25f15c15c4d3ecf21d0667.

* Fix memory errors when trim causes testcase growth

Modify trim_case_custom to avoid writing into in_buf because
some custom mutators can cause the testcase to grow rather than
shrink.

Instead of modifying in_buf directly, we write the update out
to the disk when trimming is complete, and then the caller is
responsible for refreshing the in-memory buffer from the file.

This is still a bit sketchy because it does need to modify q->len in
order to notify the upper layers that something changed, and it could
end up telling upper layer code that the q->len is *bigger* than
the buffer (q->testcase_buf) that contains it, which is asking
for trouble down the line somewhere...

* Fix an unlikely situation

Put back some `unlikely()` calls that were in
the e9d2f72382cab75832721d859c3e731da071435d commit that was
reverted.

vanhauser-thc committed:
typo

SecNotice committed:
Exit on time (#904)

* Variable AFL_EXIT_ON_TIME description has been added.
Variables AFL_EXIT_ON_TIME and afl_exit_on_time has been added.
afl->exit_on_time variable initialization has been added.
The asignment of a value to the afl->afl_env.afl_exit_on_time variable from
environment variables has been added.
Code to exit on timeout if new path not found has been added.

* Type of afl_exit_on_time variable has been changed.
Variable exit_on_time has been added to the afl_state_t structure.

* Command `export AFL_EXIT_WHEN_DONE=1` has been added.

* Millisecond to second conversion has been added.
Call get_cur_time() has been added.

* Revert to using the saved current time value.

* Useless check has been removed.

hexcoder- committed:
fix new path to custom-mutators

vanhauser-thc committed:
ensure crashes/README.txt exists

vanhauser-thc committed:
fix

WorksButNotTested committed:
Changes to bump FRIDA version and to clone FRIDA repo in to build directory rather than use a submodule as the FRIDA build scripts don't like it (#906)

Co-authored-by: Your Name <you@example.com>

WorksButNotTested committed:
Fix numeric overflow in cmplog implementation (#907)

Co-authored-by: Your Name <you@example.com>

domenukk committed:
testcase fixes for unicorn

domenukk committed:
Merge branch 'dev' of github.com:AFLplusplus/AFLplusplus into dev

hexcoder- committed:
remove merge conflict artifacts

vanhauser-thc committed:
fix afl-plot

WorksButNotTested committed:
Changes to remove binaries from frida_mode (#913)

Co-authored-by: Your Name <you@example.com>

WorksButNotTested committed:
Frida cmplog fail fast (#914)

* Changes to remove binaries from frida_mode

* Changes to make cmplog fail fast

Co-authored-by: Your Name <you@example.com>

vanhauser-thc committed:
afl-plot: relative time

vanhauser-thc committed:
arch linux and mac os support for afl-system-config

hexcoder- committed:
typo

vanhauser-thc committed:
code-format

vanhauser-thc committed:
update documentation
